mobilecoind/api/proto/mobilecoind_api.proto

```diff
@@ -338,6 +338,39 @@ message ProcessedTxOut {
     uint64 token_id = 8;
 }
 
+
+// Recoverable Transaction History memo with an optional u64 specifying the
+// subaddress index to generate the sender memo credential from.
+// Defaults to the default subaddress of the monitor.
+// Allows optinally speciying a payment intent id or payment request id.
+message TransactionMemo_RTH {
+    optional uint64 subaddress_index = 1;
```

Comment:
if this integer is optional, but defaults to zero, then maybe it's simpler for it not to be optional, since that will be the protobuf semantic anyways if it is omitted?

nbd either way